app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ReplyForwardHeaderManager.kt

```diff
@@ -0,0 +1,183 @@
+
+    fun createForwardHeader(message: Message, attachmentsToForward: List<Attachment>): String {
+        val previousBody = getHtmlDocument(message)?.let { document ->
+            val attachmentsMap = message.attachments.associate { oldAttachment ->
+                val newAttachment = attachmentsToForward.find { it.originalContentId == oldAttachment.contentId }
+
+                oldAttachment.contentId to newAttachment?.contentId
+            }
+
+            document.doOnHtmlImage { imageElement ->
+                attachmentsMap[getCid(imageElement)]?.let { newContentId ->
+                    imageElement.attr(SRC_ATTRIBUTE, "${CID_PROTOCOL}$newContentId")
+                }
+            }
+
+            return@let document.outerHtml()
+        } ?: ""
+
+        val previousFullBody = computePreviousFullBody(previousBody, message)
+
+        return assembleForwardHtmlHeader(message, previousFullBody)
+    }
+
+    fun createReplyHeader(message: Message): String {
+        val date = message.date.toDate()
+        val from = message.fromName()
+        val messageReplyHeader = appContext.getString(R.string.messageReplyHeader, date, from)
+
+        val previousBody = getHtmlDocument(message)?.let { document ->
+            val attachmentsMap = message.attachments.associate { it.contentId to it.name }
+
+            document.doOnHtmlImage { imageElement ->
+                attachmentsMap[getCid(imageElement)]?.let { name ->
+                    imageElement.replaceWith(TextNode("<$name>"))
+                }
+            }
+
+            return@let document.outerHtml()
+        } ?: ""
+
+        val previousFullBody = computePreviousFullBody(previousBody, message)
+
+        return assembleReplyHtmlHeader(messageReplyHeader, previousFullBody)
+    }
+
+    private fun Message.fromName(): String {
+        return sender?.quotedDisplay() ?: appContext.getString(R.string.unknownRecipientTitle)
+    }
+
+    private fun getHtmlDocument(message: Message): Document? {
+        val html = message.body?.let { body ->
+            when (body.type) {
+                Utils.TEXT_PLAIN -> SharedUtils.createHtmlForPlainText(body.value)
+                else -> body.value
+            }
+        }
+
+        return html?.let(Jsoup::parse)
+    }
+
+    private fun Document.doOnHtmlImage(actionOnImage: (Element) -> Unit) {
+        select(CID_IMAGE_CSS_QUERY).forEach { imageElement -> actionOnImage(imageElement) }
+    }
+
+    private fun getCid(imageElement: Element) = imageElement.attr(SRC_ATTRIBUTE).removePrefix(CID_PROTOCOL)
+
+    private fun computePreviousFullBody(previousBody: String, message: Message): String {
+        return message.body?.let { body ->
+            MessageBodyUtils.mergeSplitBodyAndSubBodies(previousBody, body.subBodies, message.uid)
+        } ?: previousBody
+    }
+
+    private fun Recipient.quotedDisplay(): String = "${("$name ").ifBlank { "" }}<$email>"
```

Comment:
Why not checking if name is blank directly ?
Or maybe checking is trimmed name is Empty ?